### PR TITLE
Migrate to Storybook 7

### PR DESCRIPTION
**Pull request by gabro**
Kind of a huge PR, but fairly mechanical. Let's see what Chromatic has to say

### CODE REVIEW

File: packages/storybook/.storybook/main.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  stories: [{ directory: "../stories", files: "**/*.stories.*" }],
+  addons: ["@storybook/addon-links", "@storybook/addon-essentials", "storybook-addon-themes"],
+  framework: {
+    name: "@storybook/react-webpack5",
+    options: {},
+  },
+  webpackFinal: (baseConfig) => {
+    const { module: { rules = [] } = {}, plugins = [] } = baseConfig;
+
+    // by default, the rules for jsx|tsx files are including the whole monorepo root,
```
Comment (veej) [inline packages/storybook/.storybook/main.ts:17]: is this still true? Does it make sense to check if we can remove these lines?

File: packages/storybook/.storybook/preview.tsx [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    const onChange = (value: unknown) => {
+      (ctx.args.onChange as (value: unknown) => void)?.(value);
+
+      // Check if the component is controlled
+      if (ctx.args.value !== undefined) {
+        setArgs({ value });
+      }
+    };
+
+    return <Story args={{ ...ctx.args, onChange }} />;
```
Comment (veej) [inline packages/storybook/.storybook/preview.tsx:27]: Couldn't be useful to allow to override the `onChange` via args?

**Review by veej — APPROVED** [review]
Looks good, even if I didn't check all the stories. Just a couple of minor comments inline

File: packages/storybook/.storybook/preview.tsx (hunk 2 above)
Comment (gabro) [reply packages/storybook/.storybook/preview.tsx:27]: maybe, I'll change it

Comment (gabro) [reply packages/storybook/.storybook/preview.tsx]: done

File: packages/storybook/.storybook/main.ts (hunk 1 above)
Comment (gabro) [reply packages/storybook/.storybook/main.ts:17]: I haven't seen anything specific to the webpack config in the changelog, so I'll keep it for now

**Pull request merged**